src/smt/expand_definitions.cpp

```diff
@@ -252,27 +317,53 @@ Node ExpandDefs::expandDefinitions(
 
   AlwaysAssert(result.size() == 1);
 
-  return result.top();
+  Node res = result.top();
+
+  if (res == orig)
+  {
+    return TrustNode::null();
+  }
+  return TrustNode::mkTrustRewrite(orig, res, tpg);
 }
 
 void ExpandDefs::expandAssertions(AssertionPipeline& assertions,
                                   bool expandOnly)
 {
   Chat() << "expanding definitions in assertions..." << std::endl;
-  Trace("simplify") << "ExpandDefs::simplify(): expanding definitions"
+  Trace("exp-defs") << "ExpandDefs::simplify(): expanding definitions"
                     << std::endl;
   TimerStat::CodeTimer codeTimer(d_smtStats.d_definitionExpansionTime);
   std::unordered_map<Node, Node, NodeHashFunction> cache;
   for (size_t i = 0, nasserts = assertions.size(); i < nasserts; ++i)
   {
     Node assert = assertions[i];
-    Node expd = expandDefinitions(assert, cache, expandOnly);
-    if (expd != assert)
+    // notice we call this method with only one value of expandOnly currently,
+    // hence we maintain only a single set of proof steps in d_tpg.
+    TrustNode expd = expandDefinitions(assert, cache, expandOnly, d_tpg.get());
+    if (!expd.isNull())
     {
-      assertions.replace(i, expd);
+      Trace("exp-defs") << "ExpandDefs::expandAssertions: " << assert << " -> "
+                        << expd.getNode() << std::endl;
+      assertions.replaceTrusted(i, expd);
     }
   }
 }
 
+void ExpandDefs::setProofNodeManager(ProofNodeManager* pnm)
+{
+  if (d_tpg == nullptr)
+  {
+    d_tpg.reset(new TConvProofGenerator(pnm,
+                                        d_smt.getUserContext(),
+                                        TConvPolicy::FIXPOINT,
+                                        TConvCachePolicy::NEVER,
+                                        "ExpandDefs::TConvProofGenerator",
+                                        nullptr,
+                                        true));
+  }
+}
+
+bool ExpandDefs::isProofEnabled() const { return d_tpg != nullptr; }
```

Comment:
We never use this method. And given how this class is set up, we probably never will...